src/Compilers/CSharp/Test/Emit3/RefStructInterfacesTests.cs

```diff
@@ -16822,7 +16822,814 @@ public class Helper
+                Diagnostic(ErrorCode.ERR_FieldAutoPropCantBeByRefLike, "T").WithArguments("T").WithLocation(7, 12),
+                // (8,12): error CS8345: Field or auto-implemented property cannot be of type 'S' unless it is an instance member of a ref struct.
+                //     public S P2;
+                Diagnostic(ErrorCode.ERR_FieldAutoPropCantBeByRefLike, "S").WithArguments("S").WithLocation(8, 12),
+                // (14,19): error CS8345: Field or auto-implemented property cannot be of type 'T' unless it is an instance member of a ref struct.
+                //     public static T P3;
+                Diagnostic(ErrorCode.ERR_FieldAutoPropCantBeByRefLike, "T").WithArguments("T").WithLocation(14, 19),
+                // (15,19): error CS8345: Field or auto-implemented property cannot be of type 'S' unless it is an instance member of a ref struct.
+                //     public static S P4;
+                Diagnostic(ErrorCode.ERR_FieldAutoPropCantBeByRefLike, "S").WithArguments("S").WithLocation(15, 19)
+                );
+        }
+
+        [Fact]
+        public void AutoProperty()
+        {
+            var src = @"
+class C<T>
+    where T : allows ref struct
+{
+    public T P1 {get; set;}
+    public S P2 {get; set;}
+}
+
+ref struct S1<T>
+    where T : allows ref struct
+{
+    public static T P3 {get; set;}
+    public static S P4 {get; set;}
+}
+
+ref struct S2<T>
+    where T : allows ref struct
+{
+    public T P5 {get; set;}
+    public S P6 {get; set;}
+}
+
+ref struct S
+{
+}
+";
+
+            var comp = CreateCompilation(src, targetFramework: s_targetFrameworkSupportingByRefLikeGenerics);
+            comp.VerifyDiagnostics(
+                // (5,12): error CS8345: Field or auto-implemented property cannot be of type 'T' unless it is an instance member of a ref struct.
+                //     public T P1 {get; set;}
+                Diagnostic(ErrorCode.ERR_FieldAutoPropCantBeByRefLike, "T").WithArguments("T").WithLocation(5, 12),
+                // (6,12): error CS8345: Field or auto-implemented property cannot be of type 'S' unless it is an instance member of a ref struct.
+                //     public S P2 {get; set;}
+                Diagnostic(ErrorCode.ERR_FieldAutoPropCantBeByRefLike, "S").WithArguments("S").WithLocation(6, 12),
+                // (12,19): error CS8345: Field or auto-implemented property cannot be of type 'T' unless it is an instance member of a ref struct.
+                //     public static T P3 {get; set;}
+                Diagnostic(ErrorCode.ERR_FieldAutoPropCantBeByRefLike, "T").WithArguments("T").WithLocation(12, 19),
+                // (13,19): error CS8345: Field or auto-implemented property cannot be of type 'S' unless it is an instance member of a ref struct.
+                //     public static S P4 {get; set;}
+                Diagnostic(ErrorCode.ERR_FieldAutoPropCantBeByRefLike, "S").WithArguments("S").WithLocation(13, 19)
+                );
+        }
```

Comment:
>} [](http://example.com/codeflow?start=8&length=1)

Consider testing capturing primary constructor parameter of type `allows ref struct`.